src/boot/bootctl-set-efivar.c

```diff
@@ -28,15 +32,29 @@ static int parse_timeout(const char *arg1, char16_t **ret_timeout, size_t *ret_t
                 timeout = USEC_INFINITY;
         else if (streq(arg1, "menu-hidden"))
                 timeout = 0;
-        else {
+        else if (streq(arg1, "menu-disabled")) {
+                uint64_t loader_features = 0;
+
+                (void) efi_loader_get_features(&loader_features);
+                if (!(loader_features & EFI_LOADER_FEATURE_MENU_DISABLE)) {
+                        if (!arg_graceful)
+                                return log_error_errno(-ENOENT, "Loader does not support 'menu-disabled': %m");
```

Comment:
the first arg of log_error_errno() and related codes, can be negative or positive, the code internally converts this to -abs(…) anyway. Thus the `-` can be dropped.

That said, please just use `SYNTEHTIC_ERRNO(ENOENT)` here, it's how we mark error numbers that we make up, where we make them up. i.e. when we pass errors we receive along we pass them as-is, but if we invent them, we do the SYNTHETIC_ERRNO() thing. This one is the latter.

Reply:
should actually be SYNTHETIC_ERRNO(EOPNOTSUPP)...